### PR TITLE
chore(bazel): add MODULE.bazel files for bzlmod

### PR DESCRIPTION
**Pull request by mmorel-35**
Commit Message: Initialisation of bzlmod, it's disabled by default with --noenable_bzlmod 
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] Related to #29849
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -58,6 +58,7 @@ test --test_verbose_timeout_warnings
 
 # Allow tags to influence execution requirements
 common --experimental_allow_tags_propagation
+common --noenable_bzlmod
```
Comment (mmorel-35) [inline .bazelrc:61]: @phlax ,
It doesn't seams to apply during ci . Any idea where it would be appropriate to apply it ?

Comment (phlax) [reply .bazelrc:61]: it should work - its the correct place to do this - which ci?

Comment (phlax) [reply .bazelrc:61]: also - im thinking it should be disabled by default in our current bazel version

Comment (mmorel-35) [reply .bazelrc:61]: > it should work - its the correct place to do this - which ci?

That's what is advised by the guide : https://bazel.build/external/migration#enable-bzlmod

Comment (mmorel-35) [reply .bazelrc:61]: > also - im thinking it should be disabled by default in our current bazel version

I think so but depending on bazel version, it's changing

Comment (mmorel-35) [reply .bazelrc:61]: > which ci? 

The azure pipeline . Bazelist is by default using the last version of bazel which is 7.0.2 . In that version bzlmod is enabled by default.
what I could see in the pipeline is  a --norc argument which can make the config not being applied for that stage

Comment (phlax) [reply .bazelrc:61]: it should be reading `.bazelversion` - do you have a link?

Comment (phlax) [reply .bazelrc:61]: also not seeing `norc` with `git grep` so wondering where that could come from

Comment (mmorel-35) [reply .bazelrc:61]: See https://dev.azure.com/cncf/envoy/_build/results?buildId=163900&view=logs&j=c7de00aa-b0fb-52ba-707b-da6afd918eda&t=449299ab-9da3-5c6b-afe3-e0c5dbf1dfd1

> /usr/bin/bash --noprofile --norc /home/vsts/work/_temp/272decbb-aa73-4860-9316-a0f8feefd019.sh

Comment (phlax) [reply .bazelrc:61]: i see - i think that is unrelated - its an argument to bash not to read (eg) rc files in home dir

Comment (phlax) [reply .bazelrc:61]: the bazel version is 6.5.0 on that ci https://source.cloud.google.com/results/invocations/cd77d311-709d-4713-8fd5-c8b2f93c8ce4/details

**Comment by mmorel-35** [comment]
Hi @alexeagle,
As you seem to be quite experienced with bzlmod, would you have some first guidance to provide here ?
How do we provide as basic fonctional MODULE organization here as there are native.binding, repository rules, etc...


Thank you !

**Comment by alexeagle** [comment]
Hi, yeah I've volunteered on a few bzlmod migrations, but I don't have time to advise on this one in my evenings/weekends. You could try other folks in the #bzlmod channel on Bazel slack. My company Aspect does  have a paid option for this kind of work.

**Pull request closed without merging**